CMakeLists.txt

```diff
@@ -9,6 +10,12 @@
 project("ubpf")
 cmake_minimum_required(VERSION 3.16)
 
+if (IS_DIRECTORY "${PROJECT_SOURCE_DIR}/.git")
+  # Install Git pre-commit hook
+  file(COPY scripts/pre-commit scripts/commit-msg
+       DESTINATION "${PROJECT_SOURCE_DIR}/.git/hooks")
+endif ()
+
```

Comment:
1. The scripts we copy are all `bash` scripts.  Does this mean we assume `bash` is available on Windows?  Or should there be PowerShell alternatives? It's also fine just to skip this if bash isn't available.
2. Would be nice to have this controllable under a Cmake option which defaults to on: `option("UBPF_INSTALL_GIT_HOOKS" "Install Git hooks" ON)`.  The CI can then turn it off explicitly.